app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -937,6 +937,6 @@ public boolean canCreateInProgramWithoutSearch() {
     private boolean displayOrgUnit() {
         return d2.organisationUnitModule().organisationUnits()
                 .byProgramUids(Collections.singletonList(currentProgram))
-                .blockingGet().size() > 1;
+                .count().blockingGet() > 1;
```

Comment:
This is a great catch!, thanks for the contribution!. Maybe these two methods could be merged into `.blockingCount()` what do you think?

Reply:
Thank you, your suggestion has a similarly good performance. Updated with a new commit.